tapiriik/services/Setio/setio.py

```diff
@@ -0,0 +1,286 @@
+        # if activity.ServiceData["Manual"]:
+        #    # Maybe implement later
+        #    activity.Laps = [Lap(startTime=activity.StartTime, endTime=activity.EndTime, stats=activity.Stats)]
+        #    return activity
+        activityID = activity.ServiceData["ActivityID"]
+
+        extID = svcRecord.ExternalID
+
+        url = "https://us-central1-project-2489250248063150762.cloudfunctions.net/getRunData"
+        payload = "{\"userId\": \"" + extID + "\",\"runId\":\"" + str(activityID) + "\"}"
+        headers = {
+            'content-type': "application/json",
+            'cache-control': "no-cache",
+        }
+        streamdata = requests.request("POST", url, data=payload, headers=headers)
+        if streamdata.status_code == 500:
+            raise APIException("Internal server error")
+
+        if streamdata.status_code == 403:
+            raise APIException("No authorization to download activity", block=True, user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+        if streamdata.status_code == 204: #"Record Not Found":
+            raise APIException("Could not find rundata")
+
+        try:
+            streamdata = streamdata.json()
+        except:
+            raise APIException("Stream data returned is not JSON")
+
+        ridedata = {}
+
+        lap = Lap(stats=activity.Stats, startTime=activity.StartTime, endTime=activity.EndTime) # Setio doesn't support laps, but we need somewhere to put the waypoints.
+        activity.Laps = [lap]
+        lap.Waypoints = []
+
+        wayPointExist = False
+
+        countWayPoints = 0
+        for stream in streamdata:
+            waypoint = Waypoint(dateutil.parser.parse(stream["time"], ignoretz=True))
+
+            if "latitude" in stream:
+                if "longitude" in stream:
+                    latitude = stream["latitude"]
+                    longitude = stream["longitude"]
+                    waypoint.Location = Location(latitude, longitude, None)
+                    if waypoint.Location.Longitude == 0 and waypoint.Location.Latitude == 0:
+                        waypoint.Location.Longitude = None
+                        waypoint.Location.Latitude = None
+
+            if "cadence" in stream:
+                waypoint.Cadence = stream["cadence"]
+                if waypoint.Cadence > 0:
+                    waypoint.Cadence = waypoint.Cadence / 2
+
+            if "elevation" in stream:
+                if not waypoint.Location:
+                    waypoint.Location = Location(None, None, None)
+                waypoint.Location.Altitude = stream["elevation"]
+
+            if "distance" in stream:
+                waypoint.Distance = stream["distance"]
+            if "speed" in stream:
+                waypoint.Speed = stream["speed"]
+            waypoint.Type = WaypointType.Regular
+            lap.Waypoints.append(waypoint)
+            countWayPoints = countWayPoints + 1

```

Comment:
Since there's a single Lap, you can just count the waypoints in it afterwards rather than maintaining a counter yourself